requestor-tutorials/service-development/service-example-2-managed-erigon.md

```diff
@@ -1,4 +1,63 @@
+---
+description: Ethereum client as a service, implemented as a self-contained runtime
+---
+
 # Service Example 2: Managed Erigon
 
-NOTE this shall **not** be included in handbook by 16th June.
+The purpose of this example is to demonstrate building a Golem service using a dedicated, self-contained runtime. It features an Ethereum network client, Erigon (f.k.a. Turbo-geth) started on provider nodes on user's demand. Interaction with the service is *not* facilitated by Golem though.
+
+{% hint style="info" %}
+This example illustrates following Golem features & aspects:
+
+* Dedicated service runtime implemented with [ya-runtime-sdk](https://github.com/golemfactory/ya-runtime-sdk)
+* Service management using [yapapi-service-manager](https://github.com/golemfactory/yapapi-service-manager)
+{% endhint %}
+
+{% hint style="info" %}
+All code for this example can be found in this repository: [https://github.com/golemfactory/yagna-service-erigon/](https://github.com/golemfactory/yagna-service-erigon/)
+{% endhint %}
+
+## Architecture overview
+
+![](erigon_architecture.jpg)
+
+The application consist of three major components:
+
+1. **User interface**  
+   A browser-based interface written in TypeScript using [React](https://reactjs.org/). It allows the end user to interact with the application, i.e. start an Erigon node after authenticating with their [Metamask](https://metamask.io/) wallet, and later manage their node(s).
+2. **Requestor server**  
+   A Python application developed using [Quart](https://pgjones.gitlab.io/quart/) and [yapapi-service-manager](https://github.com/golemfactory/yapapi-service-manager). It acts both as an HTTP server handling requests from the user interface, and as a requestor agent submitting tasks to the Golem network.
+3. **Erigon runtime**  
+    A dedicated, self-contained runtime created with [ya-runtime-sdk](https://github.com/golemfactory/ya-runtime-sdk). It is a Rust binary wrapping [Erigon](https://github.com/ledgerwatch/erigon) service itself so that it can be orchestrated by yagna daemon. The runtime also controls the access to the service by managing [Nginx](https://www.nginx.com/) configuration.
+   
+A typical interaction flow proceeds in the following manner:
+
+1. User opens the web interface in their browser.
+2. User authenticates with their metamask wallet.
+3. User clicks to start an Erigon node.
+4. Request to start an Erigon node is sent via REST API to the requestor proxy server.
```

Comment:
Few reasons why I'm against calling requestor a `proxy`:
* response is returned before any request was even sent to the provider
* `getInstances` is processed on the requestor side only
* there is a lot of communication between `yapapi` and the provider that is not caused by and requests sent to requestor

Reply:
OK, we can leave it as 'requestor server'.

Reply:
It was called 'proxy' because the end user doesn't interact directly with the Golem Network. But I'm fine with dropping this term if it suggests that requestor server is doing 1-to-1 proxying of REST API calls to the yagna daemon.